MapboxCoreNavigation/RouteController.swift

```diff
@@ -355,9 +356,14 @@ open class RouteController: NSObject {
 
 extension RouteController: Router {
     public func userIsOnRoute(_ location: CLLocation) -> Bool {
+        
+        guard let destination = routeProgress.currentLeg.destination else {
```

Comment:
Instead of defensively swallowing the `nil` destination, let’s take this opportunity to throw an assertion. If the current leg progress contains a leg that was decoded independently of any response, then I’d consider that an internal consistency error.